Reference adam and jaxsim in README (#1134)
robotology/idyntree: README.md

To avoid confusion, it is also useful to clarify what **iDynTree is not**:
* It is not the **fastest C++ library** for kinematics and dynamics multibody computations for robotics. It is not slow, but if have an application in which you need the absolute fastest library, check out [Pinocchio](https://github.com/stack-of-tasks/pinocchio).

> * It is not a **multibody simulator** library. It provides the building blocks that you could use to build a multibody simulator, but it is not a multibody simulator per se. If you need a simulator library in C++, check out [DART](https://dartsim.github.io/), [Simbody](https://github.com/simbody/simbody), [Drake](https://drake.mit.edu/), [MuJoCo](https://mujoco.org/) or the abstraction layer [`Gazebo Physics`](https://github.com/gazebosim/gz-physics). If you need a simulator implemented in MATLAB/Simulink (built on iDynTree), check [`matlab-whole-body-simulator`](https://github.com/ami-iit/matlab-whole-body-simulator). If you need a simulator that is differentiable run on GPU, check [`jaxsim`](https://github.com/ami-iit/jaxsim) or [MuJoCo XLA (mjx)](https://mujoco.readthedocs.io/en/stable/mjx.html).

Reviewer: ```suggestion
* It is not a **multibody simulator** library. It provides the building blocks that you could use to build a multibody simulator, but it is not a multibody simulator per se. If you need a simulator library in C++, check out [DART](https://dartsim.github.io/), [Simbody](https://github.com/simbody/simbody), [Drake](https://drake.mit.edu/), [MuJoCo](https://mujoco.org/) or the abstraction layer [`Gazebo Physics`](https://github.com/gazebosim/gz-physics). If you need a simulator implemented in MATLAB/Simulink (built on iDynTree), check [`matlab-whole-body-simulator`](https://github.com/ami-iit/matlab-whole-body-simulator). If you need a simulator that is differentiable and can run on GPU, check [`jaxsim`](https://github.com/ami-iit/jaxsim) or [MuJoCo XLA (mjx)](https://mujoco.readthedocs.io/en/stable/mjx.html).
```

> * It is not a **multibody simulator** library. It provides the building blocks that you could use to build a multibody simulator, but it is not a multibody simulator per se. If you need a simulator library in C++, check out [DART](https://dartsim.github.io/), [Simbody](https://github.com/simbody/simbody), [Drake](https://drake.mit.edu/), [MuJoCo](https://mujoco.org/) or the abstraction layer [`Gazebo Physics`](https://github.com/gazebosim/gz-physics). If you need a simulator implemented in MATLAB/Simulink (built on iDynTree), check [`matlab-whole-body-simulator`](https://github.com/ami-iit/matlab-whole-body-simulator). If you need a simulator that is differentiable run on GPU, check [`jaxsim`](https://github.com/ami-iit/jaxsim) or [MuJoCo XLA (mjx)](https://mujoco.readthedocs.io/en/stable/mjx.html).

Reviewer 2: ```suggestion
* It is not a **multibody simulator** library. It provides the building blocks that you could use to build a multibody simulator, but it is not a multibody simulator per se. If you need a simulator library in C++, check out [DART](https://dartsim.github.io/), [Simbody](https://github.com/simbody/simbody), [Drake](https://drake.mit.edu/), [MuJoCo](https://mujoco.org/) or the abstraction layer [`Gazebo Physics`](https://github.com/gazebosim/gz-physics). If you need a simulator implemented in MATLAB/Simulink (built on iDynTree), check [`matlab-whole-body-simulator`](https://github.com/ami-iit/matlab-whole-body-simulator). If you need a simulator that is differentiable and runs on GPU, check [`jaxsim`](https://github.com/ami-iit/jaxsim) or [MuJoCo XLA (mjx)](https://mujoco.readthedocs.io/en/stable/mjx.html).
```

> * It does not provide algorithms in a form in which they can be used in [CasADi](http://casadi.org/), [JAX](https://jax.readthedocs.io/en/latest/) or [PyTorch](https://pytorch.org/). For a Python library with an interface inspired by iDynTree that provides algorithm compatible with these frameworks, check [`adam`](https://github.com/ami-iit/adam) robotics library.

Reviewer: ```suggestion
* It does not provide algorithms in a form in which they can be used in [CasADi](http://casadi.org/), [JAX](https://jax.readthedocs.io/en/latest/) or [PyTorch](https://pytorch.org/). For a Python library with an interface inspired by iDynTree that provides algorithms compatible with these frameworks, check [`adam`](https://github.com/ami-iit/adam) robotics library.
```
